api/config/v1alpha1/envoygateway_types.go

```diff
@@ -139,11 +139,15 @@ const (
 	RedisBackendType RateLimitDatabaseBackendType = "Redis"
 )
 
-// RateLimitRedisSettings defines the configuration for connecting to
-// a Redis database.
+// RateLimitRedisSettings defines the configuration for connecting to redis database.
 type RateLimitRedisSettings struct {
 	// URL of the Redis Database.
 	URL string `json:"url"`
+
+	// TLSCertificateRef defines the client certificate reference kubernetes secret for TLS connections.
+	//
+	// +optional
+	TLSCertificateRef string `json:"tlsCertificateRef,omitempty"`
```

Comment:
I preferred the previous definition which had a top level `tls`. It allows you to add more fields in there in the future like CA cert

Reply:
https://github.com/envoyproxy/gateway/blob/ea2a688f61451b5cfebf2e23fd6a85efc9d4ea69/api/config/v1alpha1/envoygateway_types.go#L197 is a good reference

Reply:
> I preferred the previous definition which had a top level `tls`. It allows you to add more fields in there in the future like CA cert

agree, but more setting can be override by env, the CA needs different config? Imo, maybe unnecessary?

If make sure to do this, i refactor.tomorrow

Reply:
> https://github.com/envoyproxy/gateway/blob/ea2a688f61451b5cfebf2e23fd6a85efc9d4ea69/api/config/v1alpha1/envoygateway_types.go#L197
> 
> is a good reference

IMO, the `ratelimit` tls and other more fileds are different from `ExtensionTLS `, the `ratelimit` tls secret from same ns. 
so, I suggest to use the `RedisTLSSettings` directly for  the `ratelimit` extension settings.